src/hotspot/share/utilities/ostream.cpp

```diff
@@ -131,10 +131,33 @@ void outputStream::do_vsnprintf_and_write_with_automatic_buffer(const char* form
   write(str, len);
 }
 
+const char* outputStream::handle_simple_format(const char* format, va_list ap, size_t& len) {
+  const char* str = nullptr;
+  if (strchr(format, '%') == nullptr) {
+      // constant format string
+      str = format;
+      len = strlen(str);
+  } else if (format[0] == '%' && format[1] == 's' && format[2] == '\0') {
```

Comment:
Why are you assuming the %s must be at the beginning rather than "blah blah %s blah" ?

Reply:
Ah I see what you are doing now - the code just special cases the two cases of a string with no format specifiers, and a string that is only "%s". Anything else needs full handling.